### PR TITLE
blockdev: support luks

### PR DESCRIPTION
**Pull request by lolyu**
Add support of luks for `-blockdev`.

Signed-off-by: lolyu <lolyu@redhat.com>

### CODE REVIEW

**Comment by lolyu** [comment]
```
 (1/1) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.boot: PASS (51.51 s)
```

**Comment by lolyu** [comment]
```
(1/3) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (2822.92 s)
 (2/3) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.rh_kernel_update: CANCEL: The snapshot=on is not supported by -blockdev. (25.01 s)
 (3/3) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.boot: PASS (59.85 s)
```
```
```

File: virttest/qemu_devices/qcontainer.py [hunk 1]
```diff
@@ -1613,6 +1615,14 @@ def define_hbas(qtype, atype, bus, unit, port, qbus, pci_bus,
                 readonly = 'on'
             devices[-2].set_param('read-only', readonly, bool)
             devices[-1].set_param('read-only', readonly, bool)
+            if secret_obj:
+                if isinstance(devices[-1], qdevices.QBlockdevFormatQcow2):
+                    devices[-1].set_param('encrypt.format',
+                                          image_encryption.format)
+                    devices[-1].set_param('encrypt.key-secret',
+                                          secret_obj.get_qid())
+                if isinstance(devices[-1], qdevices.QBlockdevFormatLuks):
```
Comment (luckyh) [inline virttest/qemu_devices/qcontainer.py]: Suggest `if` -> `elif`

**Comment by lolyu** [comment]
@YongxueHong Please help to review, thanks.

**Comment by lolyu** [comment]
@hellohellenmao please help to review.

**Comment by hellohellenmao** [comment]
(1/3) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.unattended_install.cdrom.extra_cdrom_ks.default_install.aio_threads: PASS (1316.43 s)
 (2/3) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.rh_kernel_update: CANCEL: The snapshot=on is not supported by -blockdev. (24.23 s)
 (3/3) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.x86_64.io-github-autotest-qemu.boot: PASS (61.01 s)

**Comment by luckyh** [comment]
Thanks all.

**Pull request merged**